bonsai/BudgetTransactions/BudgetTransactions.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  BudgetTransactions.swift
+//  bonsai
+//
+//  Created by hoang on 08.12.2021.
+//
+
+import SwiftUI
+
+struct VisualEffectView: UIViewRepresentable {
+    var effect: UIVisualEffect?
+    func makeUIView(context: UIViewRepresentableContext<Self>) -> UIVisualEffectView { UIVisualEffectView() }
+    func updateUIView(_ uiView: UIVisualEffectView, context: UIViewRepresentableContext<Self>) { uiView.effect = effect }
+}
+
+struct RoundedCorner: Shape {
+
+    var radius: CGFloat = .infinity
+    var corners: UIRectCorner = .allCorners
+
+    func path(in rect: CGRect) -> Path {
+        let path = UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius))
+        return Path(path.cgPath)
+    }
+}
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
+}
+
+
+struct BudgetTransactions: View {
+    
+    private typealias TransactionGroup = [String: [Transaction]]
+    let transactions: [Transaction]
+    private let transactionsByDate: TransactionGroup
+    
+    init(transactions: [Transaction]) {
+        self.transactions = transactions
+        self.transactionsByDate = transactions
+            .reduce(into: TransactionGroup()) {
+                dict, transaction in
+                dict[transaction.date.description] = transactions
```

Comment:
.description will be different in cases with same day but different minutes/seconds/miliseconds

also .description doesnt look as designed:

<img width="371" alt="Screenshot 2021-12-12 at 22 15 32" src="https://user-images.githubusercontent.com/44097057/145726202-162c74bd-de69-4a4f-972c-22a1da8fcccf.png">
 
I suggest to write simple mapper Date -> String that will map dates like this:
if date is in today - return "Today"
yesterday - "Yesterday"
any other day - "dd.mm.yyyy"

Reply:
fixed

Reply:
👍